config/core/deployments/pingsource-mt-adapter.yaml

```diff
@@ -54,7 +54,7 @@ spec:
             - name: K_NO_SHUTDOWN_AFTER
               value: ''
             - name: K_SINK_TIMEOUT
-              value: ''
+              value: '0'
```

Comment:
Wouldn't it be more graceful to handle and empty `K_SINK_TIMEOUT` instead? E.g. default to 0 if `K_SINK_TIMEOUT == ""`

Reply:
We could but I don't think that's necessary. This is not the only env var that will cause a pod to crash is not set properly by the operator.

Reply:
I'm not too comfortable with that, and it's exactly what is called the _broken window theory_: if we have something that's already shaky there is less remorse to introduce another shaky piece to the puzzle.

Just because it's `0` in `config/` doesn't mean every user will use that manifest. Sooner or later someone will omit the var and report that same crash again.

We can make things right about `K_SINK_TIMEOUT` in particular in _this_ PR, and tackle other occurrences of unsafe env var handling in subsequent PRs. I'm happy to help if it's just a matter of time for you.

Reply:
yup agree. Fixing...